s3/main.tf

```diff
@@ -0,0 +1,26 @@
+resource "null_resource" "lazy_s3_api" {
+
+  triggers = {
+    region = var.region
+    profile = var.profile
+    bucket_name = var.bucket_name
+    owner = var.owner
+    purpose = var.purpose
+    team = var.team
+    lazy_api_host = var.lazy_api_host
+    lazy_api_key = var.lazy_api_key
+    octopus_project_space = var.octopus_project_space
+    octopus_project_name = var.octopus_project_name
```

Comment:
Why are all of these triggers? If a user changes the tag or api key, then the bucket will have to be deleted then recreated... and if there are objects in the bucket the recreate will fail.
```
Error: Error running command './cleanup.sh https://lazy.apps.ops-drivevariant.com 00o2TvfA8DjInc-RuQ1tC6nwqOnsqZ-3bsVyTuBHGK devops samir-ops-1': exit status 8. Output: Invoking create bucket end point
--2021-05-20 14:24:13--  https://lazy.apps.ops-drivevariant.com/profiles/devops/s3/samir-ops-1
Resolving lazy.apps.ops-drivevariant.com (lazy.apps.ops-drivevariant.com)... 54.245.149.241, 52.33.37.2, 44.227.111.207
Connecting to lazy.apps.ops-drivevariant.com (lazy.apps.ops-drivevariant.com)|54.245.149.241|:443... connected.
HTTP request sent, awaiting response... 409 Conflict
Saving to: ‘STDOUT’
"An error occurred (BucketNotEmpty) when calling the DeleteBucket operation: The bucket you tried to delete is not empty. You must delete all versions in the bucket."
     0K                                                       100% 39.6M=0s

2021-05-20 14:24:14 ERROR 409: Conflict.
```

Reply:
I have discussed this with vibin . They are added in triggers because any modifications done while creating should be reflected.

Reply:
What do you mean reflected? Destroyed then created again?